### ISSUE TITLE
Compiling seg3D on Ubuntu 22.04

### ISSUE THREAD

**Issue by AnoopRKulkarni** [SCIInstitute/Seg3D#418]
Hi,

I am working as part of a team handling the development of a cardiac digital twin and we need to use Seg3D for manual corrections to annotations of MRIs.

I learnt that for Linux, we need to build the software from sources. I ran into difficulty with Python_External and it would complain about _hashlib, _ssl and nis not being found. With Ubuntu 22.04 they have stopped distributing the ssl 1.0 as needed by Python 3.4.3 and without successful Python_External build, the subsequent Boost_External does not build correctly as well.

Just wanted to confirm if there is no way to compile seg3D on newer machines? Is there a workaround?

Many thanks for your advice.

Best regards
~anoop

**Comment by AnoopRKulkarni** [comment]
All,

The Seg3D-5.2.1 distribution uses a rather old Python version (3.4.3) and requires dated openSSL (1.0.2). The documentation is rather dated and inadequate.

However, I was able to successfully build Seg3D on Ubuntu 22.04 using old 1.0.2 openSSL sources. This required a few patches to the makefiles. If anyone is interested, do reach out to me and I can provide the steps I used to build Seg3D successfully.

Thanks
~anoop

**Comment by AnoopRKulkarni** [comment]
All,

The Seg3D-5.2.1 distribution uses a rather old Python version (3.4.3) and requires dated openSSL (1.0.2). The documentation is rather dated and inadequate.

However, I was able to successfully build Seg3D on Ubuntu 22.04 using old 1.0.2 openSSL sources. This required a few patches to the makefiles. If anyone is interested, do reach out to me and I can provide the steps I used to build Seg3D successfully.

Thanks
~anoop
